### PR TITLE
Add spring-cloud-dataflow-server to version-info.

### PR DESCRIPTION
**Pull request by corneil**
This adds an entry to version-info that the UI can display to ensure OSS and PRO UI shows product name and version differently.

In Pro the entry has the following:

```yaml
    spring-cloud-dataflow-server:
        name: Spring Cloud Dataflow Pro Server

```            
in OSS:

```yaml
    spring-cloud-dataflow-server:
        name: Spring Cloud Dataflow Server

```

### CODE REVIEW

File: spring-cloud-dataflow-server-core/src/main/resources/META-INF/dataflow-server-defaults.yml [hunk 1]
```diff
@@ -228,7 +231,7 @@ spring:
             - GET    /tasks/info/*                   => hasRole('ROLE_VIEW')
 
             - GET    /tasks/thinexecutions           => hasRole('ROLE_VIEW')
-
+            - GET    /tasks/thinexecutions/*         => hasRole('ROLE_VIEW')
```
Comment (cppwfs) [inline spring-cloud-dataflow-server-core/src/main/resources/META-INF/dataflow-server-defaults.yml:234]: This looks like a bug fix.  This probably needs to be in a separate PR with an explanation as to why it is needed.

Comment (onobc) [reply spring-cloud-dataflow-server-core/src/main/resources/META-INF/dataflow-server-defaults.yml:234]: Yes please

File: spring-cloud-dataflow-server-core/src/main/resources/META-INF/dataflow-server-defaults.yml [hunk 2]
```diff
@@ -94,6 +94,9 @@ spring:
         dependency-fetch:
           enabled: false
         dependencies:
+          spring-cloud-dataflow-server:
+            name: Spring Cloud Dataflow
+            version: "@project.version@"
```
Comment (onobc) [inline spring-cloud-dataflow-server-core/src/main/resources/META-INF/dataflow-server-defaults.yml:99]: Should this be `@dataflow.version@` ?

File: spring-cloud-dataflow-server-core/src/main/resources/META-INF/dataflow-server-defaults.yml [hunk 3]
```diff
@@ -94,6 +94,9 @@ spring:
...
           spring-cloud-dataflow-dashboard:
```
Comment (onobc) [inline spring-cloud-dataflow-server-core/src/main/resources/META-INF/dataflow-server-defaults.yml:100]: Where these entries in PRO required since they are the same here (OSS)? https://github.com/pivotal/scdf-pro/blob/1483cf2b965dbc4e0d7be3da3a8ada2aff6473d3/scdf-pro-server/src/main/resources/application.yml#L19-L30

**Review by onobc — CHANGES_REQUESTED** [review]
@corneil I have a couple of comments/questions. Other than that, please remove the [bug fix](https://github.com/spring-cloud/spring-cloud-dataflow/pull/5822/files#r1608390980) and create separate PR (or at the **very least** a separate commit describing what the change is).

File: spring-cloud-dataflow-server-core/src/main/resources/META-INF/dataflow-server-defaults.yml (hunk 2 above)
Comment (corneil) [reply spring-cloud-dataflow-server-core/src/main/resources/META-INF/dataflow-server-defaults.yml:99]: It will only build `spring-cloud-dataflow-server-core` in the scope of `spring-cloud-dataflow` and the project.version will be the same as `dataflow.version`

File: spring-cloud-dataflow-server-core/src/main/resources/META-INF/dataflow-server-defaults.yml (hunk 3 above)
Comment (corneil) [reply spring-cloud-dataflow-server-core/src/main/resources/META-INF/dataflow-server-defaults.yml:100]: Yes. The properties from the 2 files won't be merged.

File: spring-cloud-dataflow-server-core/src/main/resources/META-INF/dataflow-server-defaults.yml (hunk 2 above)
Comment (onobc) [reply spring-cloud-dataflow-server-core/src/main/resources/META-INF/dataflow-server-defaults.yml:99]: As long as you have tested both OSS and PRO - this is fine.

**Review by onobc — APPROVED** [review]
LGTM - thanks @corneil

**Review by cppwfs — APPROVED** [review]
